File: app/src/test/java/com/example/androidtictactoe/model/PlayerTest.java
package com.example.androidtictactoe.model;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * PlayerTest.java
 *
 * Unit testing for Player.java
 *
 * @author Chris Wolf
 * @version 1.0.0 (March 13, 2021)
 */
public class PlayerTest {
    @Test
    public void testPlayerXConstructor() {
        Player p = Player.X;
        assertEquals(p.getLabel(), "X");
    }

    @Test
    public void testPlayerOConstructor() {
        Player p = Player.O;
        assertEquals(p.getLabel(), "O");
    }

    @Test
    public void testNoPlayerConstructor() {
        Player p = Player.NONE;
        assertEquals(p.getLabel(), " ");
    }
}

File: app/src/test/java/com/example/androidtictactoe/model/PlayerTest.kt
package com.example.androidtictactoe.model

import junit.framework.Assert.assertEquals
import org.junit.Test

class PlayerTest {
    @Test
    fun testPlayerXCorrectRepresentation() {
        assertEquals("X", Player.X.representation)
    }

    @Test
    fun testPlayerOCorrectRepresentation() {
        assertEquals("O", Player.O.representation)
    }

    @Test
    fun testPlayerNoneCorrectRepresentation() {
        assertEquals(" ", Player.NONE.representation)
    }
}